### PR TITLE
Refactor health check after HCP create kubevirt

### PR DESCRIPTION
**Pull request by tiraboschi**
Remove redundant copy and pasted code:
the
"Waiting for nested cluster's node count to
reach the desired replicas count in the NodePool"
block is repeated in many steps and chains and
eventually skipped with an exit 0 first
to be executed later after proxy-setup.
Make `hypershift-kubevirt-baremetalds-proxy`
and `hypershift-kubevirt-health-check` two
plain steps (sharing the code between
MCE and not MCE chains) and reuse them
where needed with no code duplication.

### CODE REVIEW

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-master-e2e-aws-hypershift-ovn-kubevirt
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-kubevirt
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt
/pj-rehearse pull-ci-openshift-origin-release-4.17-e2e-baremetalds-kubevirt
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.17-e2e-aws-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.17-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.17-e2e-aws-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.16-e2e-aws-hypershift-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.16-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.17-e2e-aws-hypershift-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.15-e2e-aws-hypershift-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.15-e2e-aws-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-master-e2e-aws-hypershift-ovn-kubevirt
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.15-e2e-aws-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.15-e2e-aws-ovn-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-kubevirt-csi-driver-master-e2e-kubevirt-aws-csi

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-release-4.14-e2e-baremetalds-kubevirt

**Comment by tiraboschi** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-baremetalds-kubevirt

**Comment by qinqon** [comment]
I see created cluster being fine for jobs related to baremetalds so I think it's fine

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/48564/rehearse-48564-pull-ci-openshift-origin-release-4.15-e2e-baremetalds-kubevirt/1759859111597445120/artifacts/e2e-baremetalds-kubevirt/create-hostedcluster/build-log.txt

/lgtm

**Comment by tiraboschi** [comment]
Other failures are just on unrelated lanes
/pj-rehearse ack

**Comment by orenc1** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-kubevirt-aws-ovn pull-ci-openshift-hypershift-release-4.15-e2e-kubevirt-aws-ovn periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by orenc1** [comment]
The refactor works as expected and in all lanes the deployment part of the HCP has been completed successfully.
/approve

File: ci-operator/step-registry/hypershift/kubevirt/run-e2e-external/hypershift-kubevirt-run-e2e-external-chain.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ chain:
         name: release:initial
       - env: CI_HYPERSHIFT_OPERATOR
         name: pipeline:hypershift-operator
+  - ref: hypershift-kubevirt-health-check
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/hypershift/kubevirt/run-e2e-external/hypershift-kubevirt-run-e2e-external-chain.yaml:16]: this code should change to 
```
- ref: hypershift-kubevirt-install-hypershift
- ref: hypershift-kubevirt-health-check
```

Comment (tiraboschi) [reply ci-operator/step-registry/hypershift/kubevirt/run-e2e-external/hypershift-kubevirt-run-e2e-external-chain.yaml:16]: I altered `ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml` removing the redundant code from there.
This step is just after `  - chain: hypershift-kubevirt-create` one so I don't think I'm altering the previous flow.
Why do you think I should posticipate the check?

Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/kubevirt/run-e2e-external/hypershift-kubevirt-run-e2e-external-chain.yaml:16]: okay，got it

**Pull request merged**